doc/en/user/source/community/cog/mosaic.rst

```diff
@@ -0,0 +1,209 @@
+
+**Create an empty ImageMosaic without configuring it**
+
+.. admonition:: curl request
+
+   ::
+
+       curl -u admin:geoserver -XPUT --write-out %{http_code} -H "Content-type:application/zip" --data-binary @modisvi.zip http://localhost:8080/geoserver/rest/workspaces/test/coveragestores/modisvi/file.imagemosaic?configure=none
+
+*Response*
+
+::
+
+   201 OK
+
+**Providing sample prototyping granules**
+
+Next step is providing a prototype dataset for the coverage to be supported.
+
+.. admonition:: curl request
+
+   ::
+
+       curl -u admin:geoserver -XPOST -H "Content-type: text/plain" --write-out %{http_code} -d "https://modis-vi-nasa.s3-us-west-2.amazonaws.com/MOD13A1.006/2018.01.01.tif" "http://localhost:8080/geoserver/rest/workspaces/test/coveragestores/modisvi/remote.imagemosaic"
+
+*Response*
+
+::
+
+   202 Accepted
+
+
+**Initializing the store (Listing available coverages)**
+  
+Once a prototype has been provided we need to initialize the store by querying it for the available coverages.
+
+
+.. admonition:: curl request
+
+   ::
+
+       curl -v -u admin:geoserver -XGET http://localhost:8080/geoserver/rest/workspaces/test/coveragestores/modisvi/coverages.xml?list=all
+
+*Response*
+
+.. code-block:: xml
+
+    <List>
+      <coverageName>modisvi</coverageName>
+    </list>
+
+
+**Configuring the coverage**
+
+Once we get the list of available coverages, we need to configure a coverage by sending the config through REST.
+
+
+
+.. admonition:: curl request
+
+   ::
+
+       curl -v -u admin:geoserver -XPOST -H "Content-type: text/xml" -d @"coverage.xml" "http://localhost:8080/geoserver/rest/workspaces/test/coveragestores/modisvi/coverages"
+
+
+where coverage.xml has this content:
+
+.. code-block:: xml
+
+  <coverage>
+    <name>modisvi</name>
+    <nativeName>modisvi</nativeName>
+    <enabled>true</enabled>
+    <metadata>
+      <entry key="time">
+        <dimensionInfo>
+          <enabled>true</enabled>
+          <presentation>LIST</presentation>
+          <units>ISO8601</units>
+          <defaultValue>
+            <strategy>MAXIMUM</strategy>
+          </defaultValue>
+        </dimensionInfo>
+      </entry>
+    </metadata>
+  </coverage>
+
+
+**Adding more granules**
+
+Now that we have a coverageStore ready and a coverage layer configured we can start adding more granules.
+
+
+.. admonition:: curl request
+
+   ::
+
+       curl -u admin:geoserver -XPOST -H "Content-type: text/plain" --write-out %{http_code} -d "https://modis-vi-nasa.s3-us-west-2.amazonaws.com/MOD13A1.006/2018.01.17.tif" "http://localhost:8080/geoserver/rest/workspaces/test/coveragestores/modisvi/remote.imagemosaic"
+
+
+**Setting Style**
+
+That MODIS data has 2 bands representing (Normalized Difference Vegetation Index) (NDVI) and Enhanced Vegetation Index (EVI). 
+Let's add this ndvi.sld style to apply a proper colormap to the NDVI band:
+
+.. include:: src/modisvi/ndvi.sld
```

Comment:
Adding it as an in-line code block woudl have allowed to highlight syntax, like for example in the SLD cookbook:
https://raw.githubusercontent.com/geoserver/geoserver/master/doc/en/user/source/styling/sld/cookbook/points.rst

Reply:
This is what I did a the beginning indeed. Then I have started updating the sld and modifying it a bit here and there so ended up including the source. I will eventually go back to the in-line code block.